### ISSUE TITLE
REPL namespace gets reset to cljs.user when using CIDER

### ISSUE THREAD

**Issue by nybbles** [cemerick/piggieback#33]
I am experiencing this on a project based on code generated by the chestnut lein template. I would switch my CLJS repl namespace to `foo.core`. It would stay as `foo.core` until I eval'ed something, even trivial forms like `(+ 1 1)`. This would cause the CLJS repl namespace to be reset to `cljs.user`. This problem only occurred with CLJS, in CIDER and only when I use the `C-c M-n` keybinding.

CLJ REPLs work fine. The issue doesn't show up when I just use `lein repl`.

It looks like what is happening is that the REPL namespace is only being changed on Emacs and not on the browser itself. Therefore, nrepl sends back the same old `cljs.user` namespace, as can be seen by the output of `*nrepl-messages*` below.

```
(--->
  ns  "foo.core"
  op  "eval"
  session  "c935c36d-53d0-41e1-99d7-4abfff10d581"
  code  "(+ 1 1)\n"
  id  "17"
)
(<-
  id  "17"
  ns  "cljs.user"
  session  "c935c36d-53d0-41e1-99d7-4abfff10d581"
  value  "2"
)
```

If I explicitly set my namespace using `(in-ns 'foo.core)`, this problem does not occur.

**Comment by cemerick** [comment]
I think this might be related to the issue described in clojure-emacs/cider#909.

**Comment by gdanov** [comment]
this bug has a nasty effect for me. when I am loading a namespace from my project all imports are broken. example
```
(ns t.t
		(:require [clojure.string :as ss]))

(def tt[] (ss/split-lines "one\ntwo"))
```
tt gets declared in cljs.user and cannot see the import. The bigger problem is that for me even (in-ns 't.t) does not work

**Comment by cemerick** [comment]
The work on the `new-cljs-repl` branch, currently available in `0.2.0-SNAPSHOT`, is effectively a rewrite due to the new ClojureScript REPL API, etc. Closing all issues filed against `0.1.x`; most (or all) of them should be resolved and/or irrelevant.
